File: Android/app/src/main/java/alexa4/friendphoto/models/Friend.java
package alexa4.friendphoto.models;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Basic model that contains information about one friend
 */
public class Friend {
    // Default values of friend
    public final int mId;
    public final String mName;
    public final String mLastName;
    public final String mPhotoUrl; // url of profile photo
    public final String mStatus;

    /**
     * Default constructor.
     * Uses to restore from DB
     */
    public Friend(int id, String name, String lastName,
                  String photoUrl, String status) {
        mId = id;
        mName = name;
        mLastName = lastName;
        mPhotoUrl = photoUrl;
        mStatus = status;
    }

    // Constructor to restore data from JSON
    public Friend(JSONObject json) throws JSONException {
        mId = json.getInt("id");
        mStatus = json.getString("status");
        mName = json.getString("first_name");
        mLastName = json.getString("last_name");
        mPhotoUrl = json.getString("photo_100");
    }
}

File: Flutter/friend_photo_flutter/lib/src/models/friend.dart
/// Basic model that contains information about one friend
class Friend {
  // Default values of friend
  final int id;
  final String name;
  final String lastName;
  final String photoUrl; // url of profile photo
  final String status;

  ///Default constructor.
  ///Uses to restore from DB
  const Friend({this.id, this.name, this.lastName, this.photoUrl, this.status});

  /// Constructor to restore data from JSON
  factory Friend.fromJson(Map<String, dynamic> json) {
    return Friend(
      id: json['id'],
      name: json['first_name'],
      lastName: json['last_name'],
      photoUrl: json['photo_100'],
      status: json['status'],
    );
  }

  /// Convert friend object to Json. Needs to set to DB
  Map<String, dynamic> toJson() => {
        'id': id,
        'first_name': name,
        'last_name': lastName,
        'photo_100': photoUrl,
        'status': status
      };
}
